src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/package.scala

```diff
@@ -14,37 +17,43 @@ package object config {
   }
 
   implicit val dataprocConfigReader: ValueReader[DataprocConfig] = ValueReader.relative { config =>
-    DataprocConfig(config.getString("applicationName"),
-      config.getString("serviceAccount"),
+    DataprocConfig(
+      config.getString("applicationName"),
+      GoogleServiceAccount(config.getString("serviceAccount")),
       config.getString("dataprocDefaultRegion"),
-      config.getString("leoGoogleBucket"),
+      GoogleProject(config.getString("leoGoogleBucket")),
       config.getString("dataprocDockerImage"),
       config.getString("jupyterProxyDockerImage"),
-      config.getString("clusterFirewallRuleName"),
-      config.getString("clusterFirewallVPCNetwork"),
-      config.getString("clusterNetworkTag"),
-      config.getString("configFolderPath"),
-      config.getString("initActionsFileName"),
-      config.getString("clusterDockerComposeName"),
-      config.getString("leonardoServicePemName"),
-      config.getString("jupyterServerCrtName"),
-      config.getString("jupyterServerKeyName"),
-      config.getString("jupyterRootCaPemName"),
-      config.getString("jupyterRootCaKeyName"),
-      config.getString("proxySiteConf"),
       config.getString("clusterUrlBase"),
       config.getString("jupyterServerName"),
-      config.getString("proxyServerName"),
-      config.getString("jupyterInstallExtensionScript"),
-      config.getString("userServiceAccountCredentials"))
+      config.getString("proxyServerName"))
+  }
+
+  implicit val clusterResourcesConfigReader: ValueReader[ClusterResourcesConfig] = ValueReader.relative { config =>
+    val filePrefix = config.getString("configFolderPath")
+
+    ClusterResourcesConfig(
+      new File(filePrefix, config.getString("initActionsFileName")),
+      new File(filePrefix, config.getString("clusterDockerComposeName")),
+      new File(filePrefix, config.getString("leonardoServicePemName")),
+      new File(filePrefix, config.getString("jupyterServerCrtName")),
+      new File(filePrefix, config.getString("jupyterServerKeyName")),
+      new File(filePrefix, config.getString("jupyterRootCaPemName")),
+      new File(filePrefix, config.getString("jupyterRootCaKeyName")),
+      new File(filePrefix, config.getString("proxySiteConf")),
+      new File(filePrefix, config.getString("jupyterInstallExtensionScript")),
+      new File(filePrefix, config.getString("userServiceAccountCredentials")))
```

Comment:
I think you probably want to just leave this as the strings rather than file-ifying them here. There's a bunch of places where you have to then call `.getName` on the files (e.g. building bucket paths); the only time you need them to be real files on disk is when you upload them.

Reply:
Idk, I like them as `File`s because that's what they are: files on the Leo filesystem. We call `.getName` when generating paths to populate the init bucket, but that's sorta business logic. One could imagine doing other things with these files, like throwing an exception at boot time if `file.exists() == false`.

Also again, this will likely change post-GAWB-2625 because they won't be files anymore; they'll be resources inside the Leo jar.